test/haxesharp/collections/DictionaryTest.hx

```diff
@@ -0,0 +1,253 @@
+            3 => 3
+        ]);
+    }
+
+    @Test
+    public function dictionaryAllowsStringKeys()
+    {
+        testDictionaryGetSetAndRemove("foo", "bar", "key that doesn't exist");
+        testDictionaryMapConstructor([
+            "1" => 1,
+            "2" => 2,
+            "3" => 3
+        ]);
+    }
+
+    @Test
+    public function dictionaryAllowsEnumValueKeys()
+    {
+        testDictionaryGetSetAndRemove(TestEnum.OneValue, 1, TestEnum.TwoValue);
+        testDictionaryMapConstructor([
+            TestEnum.OneValue => 1,
+            TestEnum.TwoValue => 2,
+            TestEnum.ThreeValue => 3
+        ]);
+    }
+
+    @Test
+    public function dictionaryAllowsObjectKeys()
+    {
+        // dictionary uses reference equality
+        testDictionaryGetSetAndRemove(new TestObject(), 1, new TestObject());
+        var obj1 = new TestObject();
+        var obj2 = new TestObject();
+        var obj3 = new TestObject();
+        testDictionaryMapConstructor([
+            obj1 => 1,
+            obj2 => 2,
+            obj3 => 3
+        ]);
+    }
+
+    @Test
+    public function getThrowsKeyNotFoundExceptionWhenKeyDoesNotExist()
+    {
+        var testKeyNotFoundDictionary = new Dictionary<Int,Int>();
+        testKeyNotFoundDictionary.set(1, 1);
+
+        try
+        {
+            testKeyNotFoundDictionary.get(2);
+            Assert.fail("testKeyNotFoundDictionary.get(2) did not throw!");
+        }
+        catch (e:KeyNotFoundException)
+        {
+            //Pass
+        }
+    }
+
+    @Test
+    public function addThrowsArgumentExceptionOnDuplicateKey()
+    {
+        var key = 1;
+        var value = 1;
+        var testAdd = new Dictionary<Int,Int>();
+        testAdd.add(key, value);
+
+        try
+        {
+            testAdd.add(key, value);
+            Assert.fail('2nd Dictionary.add(${key}, ${value}) did not throw!');
+        }
+        catch (e:ArgumentException)
+        {
+            //Pass
+        }
+    }
+
+    @Test
+    public function setThrowsArgumentNullExceptionOnNullKey()
+    {
+        var dictionary = new Dictionary<Int,Int>(); 
+
+        try
+        {
+            dictionary.set(null, 1);
+        }
+        catch (e:ArgumentNullException)
+        {
+            //Pass
+        }
+    }
+
+    @Test
+    public function getThrowsArgumentNullExceptionOnNullKey()
+    {
+        var dictionary = new Dictionary<Int,Int>(); 
+
+        try
+        {
+            dictionary.get(null);
+        }
+        catch (e:ArgumentNullException)
+        {
+            //Pass
+        }
+    }
+
+    @Test
+    public function containsKeyThrowsArgumentNullExceptionOnNullKey()
+    {
+        var dictionary = new Dictionary<Int,Int>(); 
+
+        try
+        {
+            dictionary.containsKey(null);
+        }
+        catch (e:ArgumentNullException)
+        {
+            //Pass
+        }
+    }
+
+    @Test
+    public function removeThrowsArgumentNullExceptionOnNullKey()
```

Comment:
I don't see a test for "remove removes the object" (verify via `get`), is that test missing?

Reply:
Good point, I a bit of code to ensure that `containsKey` returns false after the specified `key` is removed.